classes/db/DbPDO.php

```diff
@@ -392,29 +392,29 @@ public static function checkSelectPrivilege($server, $user, $pwd, $db, $prefix,
             return false;
         }
 
-        if ($engine === null) {
-            $engine = 'MyISAM';
+        $enginesToTest = ['InnoDB', 'MyISAM'];
+        if ($engine !== null) {
+            $enginesToTest = [$engine];
         }
 
-        // Create a table
-        $link->query('
-		CREATE TABLE `' . $prefix . 'test` (
-			`test` tinyint(1) unsigned NOT NULL
-		) ENGINE=' . $engine);
-
-        // Select content
-        $result = $link->query('SELECT * FROM `' . $prefix . 'test`');
+        foreach ($enginesToTest as $engineToTest) {
+            $link->query('
+            CREATE TABLE `' . $prefix . 'test` (
+                `test` tinyint(1) unsigned NOT NULL
+            ) ENGINE=' . $engineToTest);
 
-        // Drop the table
-        $link->query('DROP TABLE `' . $prefix . 'test`');
+            $result = $link->query('SELECT * FROM `' . $prefix . 'test`');
 
-        if (!$result) {
-            $error = $link->errorInfo();
+            if ($result) {
+                $link->query('DROP TABLE `' . $prefix . 'test`');
```

Comment:
It should be dropped even if the `$result` failed.
What if you have the right to create but not to select.

Reply:
Nice catch 👍

Reply:
right, fixed in 676b9d3